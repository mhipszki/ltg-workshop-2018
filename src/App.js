import React, { Component } from 'react';
import logo from './jest-logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to our Jest workshop</h1>
        </header>
        <p className="App-intro">
          <a href="http://jestjs.io/en/">Jest</a> provides a fantastic user
          experience, with instant feedback and remarkable performance,
          out-of-the-box mocking, code coverage support, sandboxing, etc., as
          well as running tests in parallel. It also has a special feature:
          snapshot testing and also plays nicely with other testing libraries
          beside being highly extendable via its plugin system.
        </p>
        <p className="App-intro">
          <a href="https://pptr.dev/">Puppeteer</a> is a Node library which
          provides a high-level API to control headless Chrome or Chromium over
          the DevTools Protocol. It can also be configured to use full
          (non-headless) Chrome or Chromium.
        </p>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
