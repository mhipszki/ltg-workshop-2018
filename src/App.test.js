import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import packageJson from '../package.json';

describe('example application', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

test('says Jest workshop', () => {
  // create a new DOM element
  const div = document.createElement('div');
  // render our application and mount that to the new element
  ReactDOM.render(<App />, div);
  // attach then new element to the DOM to make it available for testing
  document.body.appendChild(div);
  // find the title
  const title = document.querySelector('h1');
  // assert on the content of the title
  expect(title.innerHTML).toContain('Jest');
});

const selector = id => `[data-test-id="${id}"]`;

test('says Jest in the title', () => {
  // create a new DOM element
  const div = document.createElement('div');
  // render our application and mount that to the new element
  ReactDOM.render(<App />, div);
  // attach then new element to the DOM to make it available for testing
  document.body.appendChild(div);
  // find the title
  const title = document.querySelector(selector('page-title'));
  // assert on the content of the title
  expect(title.innerHTML).toContain('Jest');
});

const transform = data => ({
  ...data,
  version: `version ${data.version}`
});

test('transforms package.json contents to our needs', () => {
  expect(transform(packageJson)).toMatchSnapshot();
});

test('transforms package.json contents to our needs', () => {
  expect(transform(packageJson)).toEqual({
    browserslist: {
      development: [
        'last 2 chrome versions',
        'last 2 firefox versions',
        'last 2 edge versions'
      ],
      production: ['>0.25%', 'not op_mini all', 'ie 11']
    },
    dependencies: {
      react: '^16.4.1',
      'react-dom': '^16.4.1',
      'react-scripts': '^2.0.0-next.3e165448'
    },
    devDependencies: {
      'jest-puppeteer': '^3.2.1',
      puppeteer: '^1.5.0'
    },
    name: 'workshop',
    private: true,
    scripts: {
      build: 'react-scripts build',
      eject: 'react-scripts eject',
      start: 'react-scripts start',
      test: 'react-scripts test --env=jsdom',
      'test-e2e': 'jest -c e2e-tests/jest.config.js'
    },
    version: 'version 0.1.0'
  });
});
