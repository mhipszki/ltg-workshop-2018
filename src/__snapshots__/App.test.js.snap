// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms package.json contents to our needs 1`] = `
Object {
  "browserslist": Object {
    "development": Array [
      "last 2 chrome versions",
      "last 2 firefox versions",
      "last 2 edge versions",
    ],
    "production": Array [
      ">0.25%",
      "not op_mini all",
      "ie 11",
    ],
  },
  "dependencies": Object {
    "react": "^16.4.1",
    "react-dom": "^16.4.1",
    "react-scripts": "^2.0.0-next.3e165448",
  },
  "devDependencies": Object {
    "jest-puppeteer": "^3.2.1",
    "puppeteer": "^1.5.0",
  },
  "name": "workshop",
  "private": true,
  "scripts": Object {
    "build": "react-scripts build",
    "eject": "react-scripts eject",
    "start": "react-scripts start",
    "test": "react-scripts test --env=jsdom",
    "test-e2e": "jest -c e2e-tests/jest.config.js",
  },
  "version": "version 0.1.0",
}
`;
